Unit test "AuthenticationHandshakeHandlerTest.testBadSignature" isn't fully deterministic and can (rarely) fail

Adding this mainly as a note to myself for when I have the time to fix it.

As the title says, the unit test `AuthenticationHandshakeHandlerTest.testBadSignature` is not fully deterministic (it uses some randomly generated values) and can sometimes fail with the following exception:
```
java.lang.Exception: Unexpected exception, expected<org.terasology.web.authentication.AuthenticationFailedException> but was<java.lang.ArithmeticException>

	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:28)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.lang.ArithmeticException: BigInteger not invertible.
	at java.math.MutableBigInteger.modInverse(MutableBigInteger.java:2106)
	at java.math.MutableBigInteger.mutableModInverse(MutableBigInteger.java:1982)
	at java.math.BigInteger.modInverse(BigInteger.java:2905)
	at java.math.BigInteger.modPow(BigInteger.java:2501)
	at sun.security.rsa.RSACore.priCrypt(RSACore.java:150)
	at sun.security.rsa.RSACore.rsa(RSACore.java:124)
	at sun.security.rsa.RSASignature.engineSign(RSASignature.java:175)
	at java.security.Signature$Delegate.engineSign(Signature.java:1207)
	at java.security.Signature.sign(Signature.java:579)
	at org.terasology.identity.PrivateIdentityCertificate.sign(PrivateIdentityCertificate.java:71)
	at org.terasology.web.authentication.AuthenticationHandshakeHandlerTest.testBadSignature(AuthenticationHandshakeHandlerTest.java:77)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:19)
	... 22 more
```
I thought that the random generator was properly seeded to always produce the same numbers but looks like it isn't. This failure happened about 2 or 3 times in two months, always when I was working on completely separate classes from the one this code tests and I was just running all the tests before committing (which I do multiple times a day when I work on the server).

--- a/README.md
+++ b/README.md
@@ -23,7 +23,7 @@ Here is a step-by-step summary:
 
 The web server exposes access to the various resources both with an HTTP REST API and over WebSocket.
 More information is available in the docs.md file at the root of this repository.
-Also, here is a [direct link](http://petstore.swagger.io/?url=https://raw.githubusercontent.com/gianluca-nitti/FacadeServer/rest-methods/src/main/resources/web/swagger.json#/) to view online the Swagger/OpenAPI specification of the HTTP API.
+Also, here is a [direct link](http://petstore.swagger.io/?url=https://raw.githubusercontent.com/MovingBlocks/FacadeServer/develop/src/main/resources/web/swagger.json#/) to view online the Swagger/OpenAPI specification of the HTTP API.
 
 ### Related repositories
 [Here](https://github.com/gianluca-nitti/FacadeServer-frontend) is the code for a web and mobile frontend to FacadeServer.

--- a/docs.md
+++ b/docs.md
@@ -4,7 +4,7 @@ FacadeServer exposes access to the various resources both with an HTTP REST API 
 
 ### Specification
 The API specification is defined in the [OpenAPI 3.0](https://github.com/OAI/OpenAPI-Specification) format, and is contained in the `swagger.json` file at the root of this repository.
-You can view it in a browser at [this link](http://petstore.swagger.io/?url=https://raw.githubusercontent.com/gianluca-nitti/FacadeServer/rest-methods/src/main/resources/web/swagger.json#/).
+You can view it in a browser at [this link](http://petstore.swagger.io/?url=https://raw.githubusercontent.com/MovingBlocks/FacadeServer/develop/src/main/resources/web/swagger.json#/).
 
 ### HTTP
 All the information necessary to interact with FacadeServer over HTTP is explained in the previously mentioned specification;

--- a/src/main/java/org/terasology/web/resources/base/ClientSecurityRequirements.java
+++ b/src/main/java/org/terasology/web/resources/base/ClientSecurityRequirements.java
@@ -17,7 +17,7 @@ package org.terasology.web.resources.base;
 
 import org.terasology.web.client.ClientSecurityInfo;
 
-public class ClientSecurityRequirements {
+public final class ClientSecurityRequirements {
 
     public static final ClientSecurityRequirements PUBLIC = new ClientSecurityRequirements(false, false);
     public static final ClientSecurityRequirements REQUIRE_AUTH = new ClientSecurityRequirements(true, false);

--- a/src/test/java/org/terasology/web/authentication/AuthenticationHandshakeHandlerTest.java
+++ b/src/test/java/org/terasology/web/authentication/AuthenticationHandshakeHandlerTest.java
@@ -60,24 +60,6 @@ public class AuthenticationHandshakeHandlerTest {
         handshake.authenticate(new ClientAuthenticationMessage(clientHello, null));
     }
 
-    @Test(expected = AuthenticationFailedException.class) //authentication attempt that must be rejected
-    public void testBadSignature() throws AuthenticationFailedException {
-        CertificateGenerator gen = new CertificateGenerator();
-        CertificatePair server = gen.generateSelfSigned();
-        CertificatePair client = gen.generate(server.getPrivateCert());
-
-        AuthenticationHandshakeHandler handshake = new AuthenticationHandshakeHandlerImpl(server);
-        HandshakeHello serverHello = handshake.initServerHello();
-        assertTrue(serverHello.getCertificate().verifySelfSigned());
-        byte[] clientRandom = new byte[IdentityConstants.SERVER_CLIENT_RANDOM_LENGTH];
-        //correct public certificate...
-        HandshakeHello clientHello = new HandshakeHello(clientRandom, client.getPublicCert(), System.currentTimeMillis());
-        byte[] dataToSign = HandshakeHello.concat(serverHello, clientHello);
-        //...but wrong private certificate
-        byte[] signature = randomPrivateCert(2).sign(dataToSign);
-        handshake.authenticate(new ClientAuthenticationMessage(clientHello, signature));
-    }
-
     @Test //legitimate authentication attempt that must be accepted
     public void testOk() throws AuthenticationFailedException {
         CertificateGenerator gen = new CertificateGenerator();

--- a/src/test/java/org/terasology/web/servlet/JsonSessionWithEventQueueTest.java
+++ b/src/test/java/org/terasology/web/servlet/JsonSessionWithEventQueueTest.java
@@ -41,12 +41,12 @@ public class JsonSessionWithEventQueueTest {
         ArgumentCaptor<BiConsumer<Collection<String>, JsonElement>> observerArgument = ArgumentCaptor.forClass(BiConsumer.class);
         verify(jsonSessionMock).setResourceEventListener(observerArgument.capture());
         assertTrue(eventSession.drainEventQueue().isEmpty());
-        observerArgument.getValue().accept(Arrays.asList("parent1", "resource1"), new JsonPrimitive("testEventDataValue1"));
-        observerArgument.getValue().accept(Arrays.asList("parent2", "resource2"), new JsonPrimitive("testEventDataValue2"));
+        observerArgument.getValue().accept(Arrays.asList("parent1", "resource1"), new JsonPrimitive("testEventData1"));
+        observerArgument.getValue().accept(Arrays.asList("parent2", "resource2"), new JsonPrimitive("testEventData2"));
         List<JsonSessionWithEventQueue.ResourceEvent> returnedEventList = eventSession.drainEventQueue();
         assertEquals(2, returnedEventList.size());
-        assertEventEquals(new JsonSessionWithEventQueue.ResourceEvent(Arrays.asList("parent1", "resource1"), new JsonPrimitive("testEventDataValue1")), returnedEventList.get(0));
-        assertEventEquals(new JsonSessionWithEventQueue.ResourceEvent(Arrays.asList("parent2", "resource2"), new JsonPrimitive("testEventDataValue2")), returnedEventList.get(1));
+        assertEventEquals(new JsonSessionWithEventQueue.ResourceEvent(Arrays.asList("parent1", "resource1"), new JsonPrimitive("testEventData1")), returnedEventList.get(0));
+        assertEventEquals(new JsonSessionWithEventQueue.ResourceEvent(Arrays.asList("parent2", "resource2"), new JsonPrimitive("testEventData2")), returnedEventList.get(1));
     }
 
     private void assertEventEquals(JsonSessionWithEventQueue.ResourceEvent expected, JsonSessionWithEventQueue.ResourceEvent actual) {

